src/vm.rs

```diff
@@ -550,6 +532,35 @@ impl<'a, S: Storage> Vm<'a, S> {
             Err(err) => VmExecutionResult::ExecutionError(err),
         }
     }
+
+    // Apply rewards (balance increments) based on self.reward_policy
```

Comment:
```
Apply rewards (balance increments) to beneficiary accounts, etc.
```

We also base on `block_env` and `tx` so `based on self.reward_policy` is not very precise.
Also, `reward_policy` is more of a cache to avoid recalculating block-specific values, not everything needed to calculate the rewards.